grammar.js

```diff
@@ -11,22 +19,191 @@ module.exports = grammar({
+      /[+\-]?[1-9][0-9_]*(i8|i16|i32|i64|u8|u16|u32|u64)?/
     ),
 
     symbol: $ => seq(':', choice(
-      //        A-Z        _      a-z                  0-9      A-Z       _        a-z
-      /[^\x00-\x40\x5B-\x5E\x60-\x60\x7B-\x9F][^\x00-\x2F\x3A-\x40\x5B-\x5E\x60-\x60\x7B-\x9F]*[=!\?]?/,
+      identifierRegex,
       $._operator,
       seq('"', repeat(choice(/[^"]/, '\"')), '"'),
     )),
 
+    /**
+	   * @see {@link https://crystal-lang.org/reference/syntax_and_semantics/literals/char.html}
+	   */
+    char: $ => {
+      const charsThatCanBeEscaped = [
+        "'",
+        "\\", // this is supposed to represent one backslash, but we have to escape it here
+        "a",
+        "b",
+        "e",
+        "f",
+        "n",
+        "r",
+        "t",
+        "v"
+      ];
+      const escapedLiteralChars = charsThatCanBeEscaped.map(c => "'\\" + c + "'");
+      // For stuff like '\u0042' and '\u{04AF}'
+      const unicodeEscape = /'\\u{?[0-9A-Fa-f]+}?'/;
+      return choice(
+        "''", // empty char
+        ...escapedLiteralChars,
+        unicodeEscape,
+        /'[^']'/ // any single literal char
+      );
+    },
+
+    // TODO: string interpolation
     string: $ => seq('"', repeat(choice(/[^"]/, '\"')), '"'),
 
-    _operator: $ => choice( "+", "-", "*", "/", "%", "&", "|", "^", "**", ">>", "<<", "==", "!=", "<", "<=", ">", ">=", "<=>", "===", "[]", "[]?", "[]=", "!", "~", "!~", "=~",),
+    /**
+	   * @see {@link https://crystal-lang.org/reference/syntax_and_semantics/literals/array.html}
+	   */
+    array: $ => seq(
+      '[',
+      commaSep1($._expression),
+      ']'
+    ),
+
+    /**
+	   * @see {@link https://crystal-lang.org/reference/syntax_and_semantics/literals/hash.html}
+	   */
+    hash: $ => {
+      const keyToValue = seq(
+        field('key', $._expression), 
+        '=>', 
+        field('value', $._expression)
+      );
+      const nonEmptyHash = seq(
+        // TODO: handle the MyHash{"one" => 1, "two" => 2} case
+        '{',
+        commaSep1(keyToValue),
+        '}'
+      );
+      const emptyHash = seq(
+        '{',
+        '}',
+        'of',
+        /[^\s]+/, // TODO: better "type" matcher
+        '=>',
+        /[^\s]+/ // TODO: better "type" matcher
+      );
+      return choice(
+        nonEmptyHash,
+        emptyHash
+      );
+    },
 
+    // TODO: ranges
+
+    /**
+	   * @see {@link https://crystal-lang.org/reference/syntax_and_semantics/literals/regex.html}
+	   */
+    regex: $ => seq(
+      '/',
+      /([^\/]|(\\\/))*/, // Nasty, right?
+      '/'
+    ),
+
+    /**
+	   * @see {@link https://crystal-lang.org/reference/syntax_and_semantics/literals/tuple.html}
+	   */
+    tuple: $ => choice(
+      seq(
+        '{',
+        commaSep1($._expression),
+        '}'
+      ),
+      'Tuple.new'
+    ),
+
+    /**
+	   * @see {@link https://crystal-lang.org/reference/syntax_and_semantics/literals/named_tuple.html}
+     * TODO: NamedTuple type declaration
```

Comment:
This similarly requires support for type grammar, which I haven't gotten to yet (wanted to get literals more "fleshed out" before coming back around to that)